employee-rostering-frontend/src/types.ts

```diff
@@ -15,4 +15,26 @@
  */
 export type Predicate<T> = (value: T) => boolean;
 export type ReadonlyPartial<T> = { readonly [P in keyof T]?: T[P] };
-export type Sorter<T> = (a: T, b: T) => number;
\ No newline at end of file
+export type Sorter<T> = (a: T, b: T) => number;
+export type PaginationData = {
+    itemsPerPage: number;
```

Comment:
I tried configuring my IDE many, many times. I use VS Code for TypeScript. The closest I came with was `Prettier`, but that still has disagreements with ESLint. I even tried automatically running `npm run lint -- --fix` on file save, but that result in the editor and file being out of sync. I have added a husky hook to automatically run `npm lint` on `push`, so I cannot push code unless it properly formatted (unless I use `git push --no-verify`).

Reply:
Does this help? https://travishorn.com/setting-up-eslint-on-vs-code-with-airbnb-javascript-style-guide-6eb78a535ba6

Seems you should be able to simply install ESLint extension to VS Code.

Reply:
ESLint was already installed; it was not validating Typescript code (since it normally configured to only use Javascript files). But I think I got it working now (decided to look for articles about ESLint config instead of formatters, and I appeared to get it working)